include/LMS1xx/colaa.h

```diff
@@ -0,0 +1,268 @@
+  */
+  void start_measurement();
+
+  /*!
+  * @brief Stop measurements.
+  * After receiving this command LMS1xx unit stop spinning laser and measuring.
+  */
+  void stop_measurement();
+
+  /*!
+  * @brief Set scan configuration.
+  * Get scan configuration :
+  * - scanning frequency.
+  * - scanning resolution.
+  * - start angle.
+  * - stop angle.
+  * @param cfg structure containing scan configuration.
+  */
+  void set_scan_config(const ScanConfig &cfg);
+
+  /*!
+  * @brief Set scan data configuration.
+  * Set format of scan message returned by device.
+  * @param cfg structure containing scan data configuration.
+  */
+  void set_scan_data_config(const ScanDataConfig &cfg);
+
+  /*!
+  * @brief Get current scan configuration.
+  * Get scan configuration :
+  * - scanning frequency.
+  * - scanning resolution.
+  * - start angle.
+  * - stop angle.
+  * @returns scanCfg structure.
+  */
+  ScanConfig get_scan_config();
+
+  /*!
+  * @brief Save data permanently.
+  * Parameters are saved in the EEPROM of the LMS and will also be available after the device is switched off and on again.
+  *
+  */
+  void save_config();
+
+  /*!
+  * @brief Get current status of device.
+  * This should be the same for all sensor types
+  * @returns status of device.
+  */
+  CoLaAStatus::Status query_status();
+
+  /*!
+  * @brief Get current output range configuration.
+  * Get output range configuration :
+  * - scanning resolution.
+  * - start angle.
+  * - stop angle.
+  * Should be the same for all sensor types
+  * @returns scanOutputRange structure.
+  */
+  ScanOutputRange get_scan_output_range();
+
+  /*!
+  * @brief Start or stop continuous data acquisition.
+  * After reception of this command device start or stop continuous data stream containing scan messages.
+  * @param start true : start false : stop
+  */
+  void scan_continuous(bool start);
+
+  /*!
+  * @brief Receive single scan message.
+  * @param scan_data will be passed to parse_scan_data which can be overwritten by subclasses
+  * @return true if scan was read successfully, false if error or timeout. False implies that higher level
+  *         logic should take correct action such as reopening the connection.
+  */
+  bool get_scan_data(void *scan_data);
+
+protected:
+  // Command names
+  std::string LOGIN_COMMAND;
+  std::string LOGIN_USER_MAINT;
+  std::string LOGIN_USER_AUTHORIZED;
+  std::string LOGIN_USER_SERVICE;
+  std::string LOGIN_PASS_MAINT;
+  std::string LOGIN_PASS_AUTHORIZED;
+  std::string LOGIN_PASS_SERVICE;
+
+  std::string READ_SCAN_CFG_COMMAND;
+  std::string SET_SCAN_CFG_COMMAND;
+  std::string SET_SCAN_DATA_CFG_COMMAND;
+  std::string SAVE_CONFIG_COMMAND;
+
+  std::string START_MEASUREMENT_COMMAND;
+  std::string STOP_MEASUREMENT_COMMAND;
+  std::string START_CONT_COMMAND;
+
+  std::string QUERY_STATUS_COMMAND;
+
+  std::string READ_SCAN_OUTPUT_RANGE_COMMAND;
+
+  std::string START_DEVICE_COMMAND;
```

Comment:
I did not make them static to allow overwriting the strings through inheritance. This might not be required as they are fixed in the protocol though.